src/shared.ts

```diff
@@ -16,3 +16,14 @@ export interface CompletionConfig {
     }
   >;
 }
+
+export function requireDashDashSeparator(programName: string): boolean {
```

Comment:
let's name it assertDoubleDashes and do it the assertion way

so the usage would be 

```
assertDoubleDashes(...)
const extra = ctx.rawArgs.slice(ctx.rawArgs.indexOf('--') + 1);
```

read about asserts! and is the programName arg used?